src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -780,6 +785,20 @@ public Solution AddAdditionalDocument(DocumentInfo documentInfo)
             return new Solution(newState);
         }
 
+        /// <summary>
+        /// Creates a new Solution instance that contains a new compiler configuration document like a .editorconfig file.
+        /// </summary>
+        public Solution AddAnalyzerConfigDocuments(ImmutableArray<DocumentInfo> documentInfos)
+        {
+            var newState = _state.AddAnalyzerConfigDocuments(documentInfos);
+            if (newState == _state)
```

Comment:
nit: ternary?  the IDE will even help you do that!

Reply:
My plan was to simply replace all of those with a "ForkSolution" method because this pattern is used in many, many places.